src/pages/[section]/index.tsx

```diff
@@ -1,15 +1,33 @@
-import { useRouter } from "next/router";
-import { NextPage } from "next";
+import { GetServerSideProps, NextPage } from "next";
+import { dehydrate } from "@tanstack/react-query";
 import { NewsList } from "@/components/news-list/NewsList";
 import { hasSlugStringType } from "@/utils/type-guards";
+import { prefetchNews } from "@/api/news/queries";
+import { QUERY_CLIENT } from "@/constants/query-client";
 
-const SectionPage: NextPage = () => {
-  const router = useRouter();
-  const sectionName = hasSlugStringType(router.query.section)
-    ? router.query.section
+type Props = {
+  section: string;
+};
+
+export const getServerSideProps: GetServerSideProps<Props> = async (
+  context
+) => {
+  const section = hasSlugStringType(context.query.section)
+    ? context.query.section
     : "";
 
-  return <NewsList section={sectionName} />;
+  await prefetchNews(section);
+
```

Comment:
if (!sectrion) -> not found

Reply:
Implemented

Reply:
not really - you make API call with empty parameter. why?

Reply:
Implemented throwing an error when missing the slug and returning notFound for other errors in the response from the API.